### PR TITLE
[KUBOS-350] IPC Changes

### PR DESCRIPTION
**Pull request by plauche**
Changes to the ipc layer in support for Linux Telemetry
- [x] Added new functions specific to opening/closing socket connections
- [x] Added integration tests for socket functionality
- [ ] Migrate basic csp init/cleanup functions to here

Branched from and dependent on #58

### CODE REVIEW

File: ipc/source/pubsub.c [hunk 1]
```diff
@@ -43,17 +46,44 @@ csp_socket_t * kprv_server_setup(uint8_t port, uint8_t num_connections)
...
+    }
+    if ((csp_conn = csp_accept(socket, 1000)) != NULL)
+    {
+        conn->conn_handle = csp_conn;
+        return true;
     }
     return false;
 }
 
+bool kprv_server_socket_accept(csp_socket_t * socket, pubsub_conn * conn)
```
Comment (CatFreed) [inline ipc/source/pubsub.c]: What's the intended difference between using kprv_server_socket_accept and kprv_server_accept?

Comment (plauche) [inline ipc/source/pubsub.c]: The `kprv_server_accept` is just accepting a connection over a standard csp socket. The `kprv_server_socket_accept` creates a TCP socket connection and then attempts to accept a csp socket over that.

File: .clang-format [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+SpaceBeforeAssignmentOperators: true
+SpaceBeforeParens: ControlStatements
+SpaceInEmptyParentheses: false
+SpacesBeforeTrailingComments: 1
+SpacesInAngles:  false
+SpacesInContainerLiterals: true
+SpacesInCStyleCastParentheses: false
+SpacesInParentheses: false
+SpacesInSquareBrackets: false
+Standard:        Cpp03
```
Comment (kyleparrott) [inline .clang-format]: Is the mismatch in the linter cpp03 and the c99 compilation standard we use for several targets significant?

File: ipc/source/pubsub.c (hunk 1 above)
Comment (CatFreed) [inline ipc/source/pubsub.c]: Cool, ok :+1:

**Pull request merged**